ui/src/app/base/validation.ts

```diff
@@ -25,3 +27,19 @@ export const RANGE_REGEX = /^\d{1,3}(-\d{1,3})?(,\s*(\d{1,3}(-\d{1,3})?))*$/;
  * Validate tag name string (only alphanumeric, dash or underscore)
  */
 export const TAG_NAME_REGEX = /^[a-zA-Z0-9_-]+$/;
+
+/**
+ * Validate host name string (only alphanumeric or dash and does not start or
+ * end with a dash)
+ */
+export const hostnamePattern =
+  /^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])*$/;
+
+export enum HostnameValidationLabel {
```

Comment:
Might be nice to add an additional check for strings starting with a dash and return a label explaining that could be the cause of the error?

Reply:
Done.